ballerina/tests/toXml_test.bal

```diff
@@ -0,0 +1,1169 @@
+    NSRec3 rec = {bar: {baz: "2"}};
+    xml result = check toXml(rec);
+    test:assertEquals(result.toString(), 
+        string `<x:foo xmlns:x="example.com"><x:bar><baz xmlns="example2.com">2</baz></x:bar></x:foo>`);
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapJsonToXml1() returns error? {
+    map<json> data = {id: 30, customer: {name: "Asha", age: 10}};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><customer><name>Asha</name><age>10</age></customer></root>`,
+                    msg = "testMapJsonToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml1() returns error? {
+    record {int id;} data = {id: 30};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testDefaultRecordToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml2() returns error? {
+    record {int id; string name;} data = {id: 30, name: "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testDefaultRecordToXml2 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml1() returns error? {
+    map<string> data = {"id": "30"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testMapStringToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml2() returns error? {
+    map<string> data = {"id": "30", "name": "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testMapStringToXml2 result incorrect");
+}
+
+@Name {
+    value: "Customers"
+}
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+type Customer record {
+
+    @Name {
+        value: "employeeName"
+    }
+    @Attribute
+    string ns\:name;
+
+    int age;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithAnnotationToXml1() returns error? {
+    Customer data = {ns\:name: "Asha", age: 10};
+    xml result = check toXml(data);
+    test:assertEquals(result,
+                    xml `<ns:Customers xmlns:ns="http://sdf.com" ns:employeeName="Asha"><age>10</age></ns:Customers>`,
+                    msg = "testRecordWithAnnotationToXml1 result incorrect");
+}
+
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+@Name {
+    value: "Customers"
+}
+type Customer2 record {
+
+    @Name {
+        value: "employeeName"
+    }
+    @Attribute
+    string ns\:name;
+
+    int age;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithAnnotationToXml2() returns error? {
+    Customer2 data = {ns\:name: "Asha", age: 10};
```

Comment:
why we keep the prefix here ?

Reply:
I took the tests from `xmlData` module since I port the logic from. Seems like behaviour for the `attribute` is wrong. I will correct the logic and tests.

Reply:
Fixed the behaviour and the tests [Handle attribute annotation properly for toXml](https://github.com/prakanth97/module-ballerina-data-xml/pull/8/commits/2593ba135ebc499fd70d6ee75ed9f091883b561b)